frontend/catalyst/compiler.py

```diff
@@ -258,6 +262,30 @@ def lower_mhlo_to_linalg(filename):
     return new_fname
 
 
+def transform_quantum_ir(filename):
+    """Runs quantum optimizations and transformations, as well gradient transforms, on the hybrid
+    IR.
+
+    Args:
+        filename (str): the path to a file were the program is stored.
+    Returns:
+        a path to the output file
+    """
+    if filename[-5:] != ".mlir":
```

Comment:
We don't use `os.extsep` anywhere else, and in particular we don't use it when we generate the file names.

I'm not opposed to using it in general, but it would have to be consistent (see the [discussion here](https://github.com/PennyLaneAI/catalyst/pull/67#discussion_r1145368377)).

Reply:
Ah! I thought that PR #68  is already merged. This update was originally addressed there. Nevermind :)